### DISCUSSION TITLE
How to escape double underscore and single underscore in rg.

### DISCUSSION THREAD

**Discussion by jianhe-fun** [BurntSushi/ripgrep#2324] [General] [answered]
I do tried. 
```
rg '[[__FLT_DIG__]]' .
rg \_\_FLT_DIG\_\_ .
rg "__FLT_DIG__" .
rg '__FLT_DIG__' .

```
But How do I capture content contains `__FLT_DIG__`
for example, the following line do match the pattern.
` #define FLT_DIG __FLT_DIG__`

**Comment by BurntSushi** [comment, marked as answer]
This is almost certainly an instance of the [XY problem](https://en.wikipedia.org/wiki/XY_problem). Namely, underscores never need to be escaped because they are not meta characters: 

```
$ echo '#define FLT_DIG __FLT_DIG__' | rg __FLT_DIG__
#define FLT_DIG __FLT_DIG__
```

So if `rg __FLT_DIG__` isn't showing the matches you expect, then it's almost certainly because of some other problem. You'll need to either provide a complete reproduction, and/or take a look at the `--debug` output. Remember, as the very first lines of the ripgrep docs state, ripgrep will by default ignore things matching your gitignore, hidden files and binary files.
